src/c_api/c_api.cc

```diff
@@ -550,6 +566,34 @@ int MXNDArrayGetShapeEx(NDArrayHandle handle,
   API_END();
 }
 
+int MXNDArrayGetShapeExInt64(NDArrayHandle handle,
```

Comment:
If the internal implementation is the same and only differs by data type, we can create an internal template to do this?